### PR TITLE
added morris traversal algorithm

### PR DESCRIPTION
**Pull request by siddharthzs**
### **Describe your change:**
Added Morris Traversal Algorithm to traversal Directory

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by siddharthzs** [comment]
My pre-commit is failing, I am not getting why? Could please help me fix it.

**Comment by siddharthzs** [comment]
Hello @dhruvmanila ,
My changes are ready , please review them.
Thankyou

File: traversals/morris_traversal.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    return InorderTraversal
+
+
+root = Node(4)
+temp = root
+temp.left = Node(2)
+temp.right = Node(8)
+temp = temp.left
+temp.left = Node(1)
+temp.right = Node(5)
```
Comment (poyea) [inline traversals/morris_traversal.py:49]: Can you put these into the doctest of the function?

File: traversals/morris_traversal.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+print(Morris_traversal(root))
```
Comment (poyea) [inline traversals/morris_traversal.py:51]: This can also be put inside `if __name__ == "__main__":`

**Review by poyea — CHANGES_REQUESTED** [review]

File: traversals/morris_traversal.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+class Node:
+    def __init__(self, data):
+        self.data = data
+        self.left = None
+        self.right = None
+
+
+def Morris_traversal(root):
```
Comment (dhruvmanila) [inline traversals/morris_traversal.py:8]: - `snake_case` for function name
- Type hints
- Doctest required

File: traversals/morris_traversal.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    """
+    Morris(InOrder) travaersal is a tree traversal algorithm that does not employ
+     the use of recursion or a stack. In this traversal, links are created as
+    successors and nodes are printed using these links.
+    Finally, the changes are reverted back to restore the original tree.
+    """
+    InorderTraversal = []
```
Comment (dhruvmanila) [inline traversals/morris_traversal.py:15]:
```suggestion
    inorder_traversal = []
```

Comment (dhruvmanila) [reply traversals/morris_traversal.py:15]: Please do this change wherever the variable was used.

File: traversals/morris_traversal.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    # set current to root of binary tree
+    curr = root
```
Comment (dhruvmanila) [inline traversals/morris_traversal.py:17]:
```suggestion
    current = root
```
Explicit is better than implicit
Please make this change wherever the variable was used.

File: traversals/morris_traversal.py (hunk 4 above)
Comment (siddharthzs) [reply traversals/morris_traversal.py:15]: Sure, I will do

**Review by siddharthzs — COMMENTED** [review]
Sure, I do the necessary changes.

File: traversals/morris_traversal.py (hunk 3 above)
Comment (dhruvmanila) [reply traversals/morris_traversal.py]: Type hints and doctests are required. Please google it.

**Comment by siddharthzs** [comment]
Hello, @dhruvmanila, I request you to close the issue, if there are no further changes to be made.
 #2538

**Pull request closed without merging**